src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1298,13 +1300,22 @@ class ObjectGenerator : public Generator<bsoncxx::document::value> {
           _keyGen{stringGenerator(node["havingKeys"], generatorArgs)},
           _valueGen{
               valueGenerator<false, UniqueAppendable>(node["andValues"], generatorArgs, parsers)},
-          _nTimesGen{intGenerator(extract(node, "withNEntries", "^Object"), generatorArgs)} {}
+          _nTimesGen{intGenerator(extract(node, "withNEntries", "^Object"), generatorArgs)},
+          _allowDuplicateKeys{node["allowDuplicateKeys"].to<bool>()} {}
```

Comment:
I've decided against providing a default as for the users of the Object generator it might be equally unexpected when the object would contain duplicated keys _or_ would have fewer keys than requested.